### PR TITLE
Added new types for uni-app

### PR DESCRIPTION
**Pull request by zhetengbiji**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/uni-app/tslint.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "max-line-length": false,
```
Comment (sandersn) [inline types/uni-app/tslint.json]: Can you remove these overrides and fix up the resulting lint?

File: types/uni-app/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// Type definitions for non-npm package uni-app 1.0
+// Project: https://github.com/dcloudio/uni-app
+// Definitions by: DCloud <https://github.com/dcloudio>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// TypeScript Version: 2.3
+
+import "./lib/app";
```
Comment (sandersn) [inline types/uni-app/index.d.ts]: Why are these needed? Aren’t they all already global? Is it just to satisfy the linter?

**Review by sandersn — CHANGES_REQUESTED** [review]

File: types/uni-app/tslint.json (hunk 1 above)
Comment (zhetengbiji) [reply types/uni-app/tslint.json]: ok，I will adjust

File: types/uni-app/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6",
+            "dom"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": false,
```
Comment (sandersn) [inline types/uni-app/tsconfig.json]: Does something break with strictNullChecks: true? Your consumers may have it turned on, so it should ideally work that way.

Comment (zhetengbiji) [reply types/uni-app/tsconfig.json]: ok

**Review by sandersn — APPROVED** [review]

**Comment by sandersn** [comment]
Thanks for your patience on this one @zhetengbiji
